src/lib.rs

```diff
@@ -92,32 +115,67 @@ pub fn run(args: Args) {
                 args.filter_arg.clone(),
             );
 
+            printer.get_mut().clear();
```

Comment:
This is what was missing (vs what `ripgrep` [does](https://github.com/BurntSushi/ripgrep/blob/4fcb1b2202b97c5a21894672232700225223a138/crates/core/main.rs#L159))